app/code/Vladimirl/Chatter/Controller/Submit/Submit.php

```diff
@@ -57,17 +98,56 @@ public function generateHash(): string
     public function execute()
     {
         try {
-            $request = $this->getRequest();
-            $websiteId = (int)$this->storeManager->getWebsite()->getId();
-            $authorType = 'customer';
-            $authorName = 'anonymous';
+            if (!$this->formKeyValidator->validate($this->getRequest())) {
+                throw new LocalizedException(__('Something went wrong!'));
+            }
+
+            $customerId = (int) $this->customerSession->getId();
+            $websiteId = (int) $this->storeManager->getWebsite()->getId();
+
+            if (!$chatHash = $this->customerSession->getChatHash()) {
+                $this->customerSession->setChatHash($this->generateHash());
+            }
+
+            if ($customerId) {
+                $transaction = $this->transactionFactory->create();
+                $authorType = 'customer';
+                $authorName = $this->customerSession->getCustomerData()->getEmail();
+
+                $messageCollection = $this->messageCollectionFactory->create();
+                $messageCollection->addCustomerIdFilter($customerId)
+                    ->addAuthorTypeFilter($authorType);
+                $oldChatHash = $messageCollection->getFirstItem()->getChatHash();
+                if ($oldChatHash === null) {
+                    $oldChatHash = $this->customerSession->getChatHash();
+                }
+
+                $messageCollection = $this->messageCollectionFactory->create();
+                $messageCollection->addChatHashFilter($chatHash);
+
+                foreach ($messageCollection as $existingMessage) {
+                    if ($existingMessage->getAuthorId() !== $customerId) {
```

Comment:
@Vladimir-L This will not work because `::getAuthorId()` will actually return a string when the data is fetched from the database. You must type cast this to integer or implement method `::getAuthorId()` so that it really returns integer. This is bad part of how Magento models and collections are implemented - everything is a string by default until you type cast it.